File: backend/src/main/java/br/ucs/greencitizenship/entities/Post.java
package br.ucs.greencitizenship.entities;

import br.ucs.greencitizenship.entities.enums.StatusEnum;
import br.ucs.greencitizenship.entities.enums.converters.StatusEnumConverter;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Builder
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(of = "id")
@Entity
@Table(name = "tb_post")
public class Post {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User author;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

    private String title;
    private String description;
    private LocalDateTime date;
    @Convert(converter = StatusEnumConverter.class)
    private StatusEnum status;
    private Boolean isUrgent;

    @OneToOne(mappedBy = "post", cascade = CascadeType.REMOVE)
    private PostAttachment postAttachment;

    @OneToMany(mappedBy = "post", cascade = CascadeType.REMOVE, fetch = FetchType.LAZY)
    private List<Like> likes = new ArrayList<>();

    @OneToMany(mappedBy = "comment", cascade = CascadeType.REMOVE, fetch = FetchType.LAZY)
    private List<Comment> comments = new ArrayList<>();

    public Post(Integer id) {
        this.id = id;
    }
}

File: frontend/src/models/post.ts
import { DCategory } from "./category";
import { DComment } from "./comment";
import { DStatusEnum } from "./enums/statusEnum";
import { DLike } from "./like";
import { DPostAttachment } from "./postAttachment";
import { DUser } from "./user";

export type DPost = {
    id: number;
    author: DUser;
    category: DCategory;
    title: string;
    description: string;
    date: Date;
    status: keyof typeof DStatusEnum;
    isUrgent: boolean;
    postAttachment: DPostAttachment;

    likes: DLike[];
    comments: DComment[];
}